src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -80,3 +83,56 @@ void sfSoundRecorder_setProcessingInterval(sfSoundRecorder* soundRecorder, sfTim
 {
     CSFML_CALL(soundRecorder, setProcessingInterval(interval));
 }
+
+
+////////////////////////////////////////////////////////////
+const char** sfSoundRecorder_getAvailableDevices(size_t* count)
+{
+    static std::vector<std::string> stringDevices;
+    static std::vector<const char*> cstringDevices;
+
+    if (stringDevices.empty())
+    {
+        stringDevices = sf::SoundRecorder::getAvailableDevices();
+        cstringDevices.reserve(stringDevices.size());
+
+        for (std::vector<std::string>::const_iterator it = stringDevices.begin(); it != stringDevices.end(); ++it)
+        {
+            cstringDevices.push_back(it->c_str());
+        }
+    }
+
+    if (count)
+        *count = cstringDevices.size();
+
+    return !cstringDevices.empty() ? &cstringDevices[0] : NULL;
+}
+
+////////////////////////////////////////////////////////////
+const char* sfSoundRecorder_getDefaultDevice()
+{
+    static std::string defaultDevice;
+
+    if (!defaultDevice.size())
+    {
+        defaultDevice = sf::SoundRecorder::getDefaultDevice();
+    }
+
```

Comment:
If we did this then we could also shorten the device list function the same way. The only reason I didn't do this because I'm not sure when the initialization would happen. And the history of issues with global initialization order... :grin: